cfme/services/catalogs/catalog_items/__init__.py

```diff
@@ -294,18 +318,47 @@ def delete(self):
         assert view.is_displayed
         view.flash.assert_success_message('The selected Catalog Item was deleted')
 
-    def add_button_group(self):
-        button_name = fauxfactory.gen_alpha()
-        view = navigate_to(self, 'AddButtonGroup')
-        view.fill({'btn_group_text': 'group_text',
-                   'btn_group_hvr_text': button_name,
-                   'btn_image': self.button_icon_name})
+    def add_button_group(self, **kwargs):
+        button_name = kwargs.get("text", "gp_{}".format(fauxfactory.gen_alpha()))
+        hover_name = kwargs.get("hover", "hover_{}".format(fauxfactory.gen_alpha()))
+        image = kwargs.get("image", "broom")
+
+        view = navigate_to(self, "AddButtonGroup")
+        view.fill(
+            {
+                "btn_group_text": button_name,
+                "btn_group_hvr_text": hover_name,
+                "btn_image": image,
+            }
+        )
         view.add.click()
-        view = self.create_view(DetailsCatalogItemView)
-        assert view.is_displayed
+        view = self.create_view(DetailsCatalogItemView, wait="10s")
         view.flash.assert_no_error()
         return button_name
 
+    def button_group_exists(self, name):
+        view = navigate_to(self, "Details")
+        path = view.catalog_items.tree.read()
+        path.extend(["Actions", "{} (Group)".format(name)])
+        try:
+            view.catalog_items.tree.fill(path)
+            return True
+        except Exception as e:
+            logger.exception(e)
+            return False
+
+    def delete_button_group(self, name):
+        view = navigate_to(self, "Details")
+        path = view.catalog_items.tree.read()
+        path.extend(["Actions", "{} (Group)".format(name)])
+        view.catalog_items.tree.fill(path)
+        view.configuration.item_select("Remove this Button Group", handle_alert=True)
+        view.flash.assert_no_error()
+
+        # TODO(BZ-1687289): To avoid improper page landing adding a workaround.
+        if BZ(1687289, forced_streams=['5.9', '5.10']).blocks:
```

Comment:
'5.10' does not need to be forced, and we have set a breakpoint for '5.9'.

You can just use `BZ(1687289)` in this case.

```
Out[2]: Version('5.10.2.2')
In[3]: from cfme.utils.blockers import BZ
In[4]: BZ(1687289).blocks
Out[4]: True
```